webservices/common/models/reports.py

```diff
@@ -60,6 +60,32 @@ def treasurer_name(self):
         )
         return name
 
+class AmendmentChainMixin(object):
+
+    @property
+    def most_recent(self):
+        return self.file_number == self.amendment.most_recent_filing
+
+    @hybrid_property
+    def most_recent_filing(self):
+        return self.amendment.most_recent_filing
+
+    @hybrid_property
+    def amendment_chain(self):
+        return self.amendment.amendment_chain
+
+    @property
+    def is_amended(self):
+        return not self.most_recent
+
+    @property
+    def amended_by(self):
+        if len(self.amendment.longest_chain) > 0 and self.is_amended:
+            index = self.amendment.longest_chain.index(self.file_number)
```

Comment:
I'd agree that the logic is spread out to two other classes, and I accounted that by putting these behaviors in their own class and using a mixin to modify the behavior of the classes that need it.  I guess it could be done the other way, but I'm trying to match the common pattern of using mixins we've used throughout the api.